tests/integ/test_simplifier_suite.py

```diff
@@ -501,3 +501,41 @@ def test_use_sql_simplifier(session, simplifier_table):
         Utils.check_answer(df3, df4, sort=True)
     finally:
         context._use_sql_simplifier = True
+
+
+def test_select_star(session, simplifier_table):
+    df = session.table(simplifier_table)
+    df1 = df.select("*")
+    assert df1.queries["queries"][0] == f"SELECT  *  FROM {simplifier_table}"
+
+    df2 = df.select(df["*"])  # no flatten
+    assert (
+        df2.queries["queries"][0]
+        == f'SELECT "A","B" FROM ( SELECT  *  FROM {simplifier_table})'
+    )
+
+    df3 = df.select("*", "a")
+    assert (
+        df3.queries["queries"][0]
+        == f'SELECT *, "A" FROM ( SELECT  *  FROM {simplifier_table})'
+    )
+
+    df4 = df.select(df["*"], "a")
+    assert (
+        df4.queries["queries"][0]
+        == f'SELECT "A","B", "A" FROM ( SELECT  *  FROM {simplifier_table})'
+    )
+
+    df5 = df3.select("b")
+    assert (
+        df5.queries["queries"][0]
+        == f'SELECT "B" FROM ( SELECT *, "A" FROM ( SELECT  *  FROM {simplifier_table}))'
+    )
+
+    with pytest.raises(SnowparkSQLException, match="ambiguous column name 'A'"):
+        df3.select("a").collect()
+
+    with pytest.raises(SnowparkSQLException, match="ambiguous column name 'A'"):
+        df4.select("a").collect()
+
+    # tests/integ/scala/test_dataframe_join_suite.py has tests of join and select *. We don't repeat those tests here.
```

Comment:
are there test to verify the join simplifier also flattens for example joining two tables and making sure dropped columns are not flattened and multiple `.select('a').select('a')` is flattened?

Reply:
Good point. Added the test